openhands/runtime/client/runtime.py

```diff
@@ -293,53 +293,57 @@ def _init_container(
+
+            # Always process logs, regardless of client_ready status
+            logs = self.log_buffer.get_and_clear()
+            if logs:
+                formatted_logs = '\n'.join([f'    |{log}' for log in logs])
+                logger.info(
+                    '\n'
+                    + '-' * 35
+                    + 'Container logs:'
+                    + '-' * 35
+                    + f'\n{formatted_logs}'
+                    + '\n'
+                    + '-' * 80
+                )
 
-        # Always process logs, regardless of client_ready status
-        logs = self.log_buffer.get_and_clear()
-        if logs:
-            formatted_logs = '\n'.join([f'    |{log}' for log in logs])
-            logger.info(
-                '\n'
-                + '-' * 35
-                + 'Container logs:'
-                + '-' * 35
-                + f'\n{formatted_logs}'
-                + '\n'
-                + '-' * 80
-            )
+            if not self.log_buffer.client_ready:
+                time.sleep(1)
+                attempts = 0
+                while not self.log_buffer.client_ready and attempts < 5:
+                    attempts += 1
+                    time.sleep(1)
+                    logs = self.log_buffer.get_and_clear()
+                    if logs:
+                        formatted_logs = '\n'.join([f'    |{log}' for log in logs])
+                        logger.info(
+                            '\n'
+                            + '-' * 35
+                            + 'Container logs:'
+                            + '-' * 35
+                            + f'\n{formatted_logs}'
+                            + '\n'
+                            + '-' * 80
+                        )
+
+            response = self.session.get(f'{self.api_url}/alive')
+            if response.status_code == 200:
+                return
+            else:
+                msg = f'Action execution API is not alive. Response: {response}'
+                logger.error(msg)
+                raise RuntimeError(msg)
 
-        if not self.log_buffer.client_ready:
-            time.sleep(1)
-            attempts = 0
-            while not self.log_buffer.client_ready and attempts < 4:
-                attempts += 1
-                time.sleep(2)
-                logs = self.log_buffer.get_and_clear()
-                if logs:
-                    formatted_logs = '\n'.join([f'    |{log}' for log in logs])
-                    logger.info(
-                        '\n'
-                        + '-' * 35
-                        + 'Container logs:'
-                        + '-' * 35
-                        + f'\n{formatted_logs}'
-                        + '\n'
-                        + '-' * 80
-                    )
-
-        response = self.session.get(f'{self.api_url}/alive')
-        if response.status_code == 200:
+        except KeyboardInterrupt:
+            logger.debug('KeyboardInterrupt: exiting _wait_until_alive.')
```

Comment:
I'd suggest to not add this, because it doesn't work as expected. The message is not in the logs.

It will be confusing us, when we see it in the code and assume it worked. 😢

Reply:
What exactly shall I remove, just the logger message or the except?